osu.Game/Online/API/Requests/Responses/APIRecentActivity.cs

```diff
@@ -41,9 +41,38 @@ private string scoreRank
         [JsonProperty("count")]
         public int Count;
 
-        [JsonProperty("mode")]
         public string Mode;
 
+        [JsonProperty("mode")]
+        private string mode
+        {
+            set
+            {
+                switch (value)
+                {
+                    default:
+                        Mode = value;
+                        return;
+
+                    case "osu":
+                        Mode = "osu!";
+                        return;
+
+                    case "mania":
+                        Mode = "osu!mania";
+                        return;
+
+                    case "taiko":
+                        Mode = "osu!taiko";
+                        return;
+
+                    case "fruits":
+                        Mode = "osu!catch";
+                        return;
+                }
+            }
+        }
```

Comment:
This should not be a thing. The string before was *somewhat* acceptable, but now you have mixed in two magic strings in this class, one representing data returned from API and the other representing data *presentation*.

I'd either use an enum with `[Description]`s or use `RulesetInfo.ShortName` (higher up, not here directly) to get an actual ruleset object with a proper name.